x/programs/program/context.go

```diff
@@ -5,10 +5,37 @@ package program
 
 import (
 	"github.com/ava-labs/avalanchego/ids"
+	"github.com/near/borsh-go"
 )
 
 type Context struct {
 	ProgramID ids.ID `json:"program"`
+	result    []byte
 	// Actor            [32]byte `json:"actor"`
 	// OriginatingActor [32]byte `json:"originating_actor"`
 }
+
+func (c *Context) Result() []byte {
+	return c.result
+}
+
+func (c *Context) SetResult(result []byte) {
+	c.result = result
+}
+
+func (c *Context) ClearResult() {
+	c.result = nil
+}
+
+type serializeable struct {
+	ProgramID ids.ID
+}
+
+func (c *Context) WriteToMem(mem *Memory) (uint32, error) {
+	bytes, err := borsh.Serialize(serializeable{ProgramID: c.ProgramID})
```

Comment:
I know I know, this is gross... I think @dboehm-avalabs found a much better way to do this so I don't it doesn't really matter